synse/commands/fan_sensors.py

```diff
@@ -55,10 +122,56 @@
                 single_reading['location'] = {
                     'rack': rack,
                     'board': board,
-                    'device': device
+                    'device': device,
                 }
+                single_reading['scan_cache_device'] = scan_cache_device
                 logger.debug('fan_sensors data with vec: {}.'.format(single_reading))
                 readings.append(single_reading)
 
+                # If the rack is not a key in new readings, add it.
+                if rack not in new_readings['racks']:
+                    new_readings['racks'][rack] = dict()
+
+                # Translate single_reading['scan_cache_device']['info']
+                # and add it under the rack key which is:
+                # new_readings['racks'][rack][translation] \
+                #     = single_reading['data'][single_reading['type']]['value']
+                logger.debug(
+                    'single_reading[scan_cache_device][info]: {}'.format(
+                        single_reading['scan_cache_device']['info']))
+                logger.debug(
+                    'single_reading[data][single_reading[type]][value]: {}'.format(
+                        single_reading['data'][single_reading['type']]['value']))
+
+                # Add sensor reading to result set.
+                fan_sensor_key = _translate_device_info(single_reading['scan_cache_device']['info'])
+                reading_value = single_reading['data'][single_reading['type']]['value']
+                if fan_sensor_key is not None and reading_value is not None:
+                    # Be sure not to overwrite any existing reading in the current result set.
+                    # That would imply a mapping issue or some other bug.
+                    if fan_sensor_key in new_readings['racks'][rack]:
+                        message = 'fan_sensors avoiding overwrite of existing reading [{}] at ' \
+                                  'new_readings[racks][{}][{}] with [{}]'.format(
+                                      new_readings['racks'][rack][fan_sensor_key],
+                                      rack, fan_sensor_key, reading_value)
+                        logger.error(message)
+                        raise ValueError(message)
+                    # No existing reading in the result set, safe to add it.
+                    new_readings['racks'][rack][fan_sensor_key] = reading_value
+
+
     logger.debug('--- FAN SENSORS end ---')
-    return readings
+    # Sort the new_readings racks by racks['id']
+    new_readings['racks'] = OrderedDict(sorted(new_readings['racks'].items()))
```

Comment:
I'm probably just brainfarting here, but what in auto-fan requires stuff to be sorted?

Reply:
My sanity when debugging it is a good place to start. The initial result set came back like this below. Trying to assure that the sensors are aggregated properly without a sort is madness.

It's not spelled out in the result set above, but each thermistor (0-11) corresponds to the channel that it's on. That channel corresponds to the wiring diagram which corresponds to a rack location.

Pressure are similar, but index 0 is channel 1 bottom, index 1 is channel 2 middle, and index 2 is channel 4 top. 

```
curl http://${SYNSE_IP}:5000/synse/2.0/fan_sensors
[
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.49,
        "timestamp":"2018-03-27 00:27:35.275386728 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec1-c1-wrigley",
      "board":"vec",
      "device":"1e93da83dd383757474f539314446c3d"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.78,
        "timestamp":"2018-03-27 00:27:35.377430472 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec1-c1-wrigley",
      "board":"vec",
      "device":"18185208cbc0e5a4700badd6e39bb12d"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.78,
        "timestamp":"2018-03-27 00:27:35.490718481 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec1-c1-wrigley",
      "board":"vec",
      "device":"e955c02a134ff2ac6e9e3a9dedc67dc1"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.71,
        "timestamp":"2018-03-27 00:27:34.381839366 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec1-c1-wrigley",
      "board":"vec",
      "device":"2079f261767869c929e01ba10132eceb"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":null,
        "timestamp":"2018-03-27 00:27:34.493652092 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec1-c1-wrigley",
      "board":"vec",
      "device":"b4bf920c24ad7c2e23360ffcda9f4b5d"
    }
  },
  {
    "type":"pressure",
    "data":{
      "pressure":{
        "value":-665.95,
        "timestamp":"2018-03-27 00:27:35.102268176 +0000 UTC",
        "unit":{
          "symbol":"Pa",
          "name":"pascals"
        }
      }
    },
    "location":{
      "rack":"vec1-c1-wrigley",
      "board":"vec",
      "device":"285e7b428eb467678097ecab5f765d7c"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":null,
        "timestamp":"2018-03-27T00:27:36.794492+00:00",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec1-c1-wrigley",
      "board":"vec",
      "device":"3f336eb0bb188de8c72650d406225ffa"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":18.65,
        "timestamp":"2018-03-27 00:27:35.586119185 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec1-c1-wrigley",
      "board":"vec",
      "device":"b0efdac41dfc7e8261dfd4bd2369b161"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.78,
        "timestamp":"2018-03-27 00:27:35.686236839 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec1-c1-wrigley",
      "board":"vec",
      "device":"5beec8c6db5445fd164566478997ff30"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.71,
        "timestamp":"2018-03-27 00:27:35.818183615 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec1-c1-wrigley",
      "board":"vec",
      "device":"fa941bad2325cb097deb97743007ff36"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":null,
        "timestamp":"2018-03-27 00:27:35.923367447 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec1-c1-wrigley",
      "board":"vec",
      "device":"c5d8272bc1106aa12225850d33483f33"
    }
  },
  {
    "type":"pressure",
    "data":{
      "pressure":{
        "value":9.764,
        "timestamp":"2018-03-27 00:27:36.184960331 +0000 UTC",
        "unit":{
          "symbol":"Pa",
          "name":"pascals"
        }
      }
    },
    "location":{
      "rack":"vec1-c1-wrigley",
      "board":"vec",
      "device":"880d01fd6edd947ceb75f131324eecb7"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.85,
        "timestamp":"2018-03-27 00:27:36.311018694 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec1-c1-wrigley",
      "board":"vec",
      "device":"80de4a2b73c41cf71ef7d30e1f2b9494"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.93,
        "timestamp":"2018-03-27 00:27:36.446805484 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec1-c1-wrigley",
      "board":"vec",
      "device":"645a790d97fc93ee12ec5fbb514230d7"
    }
  },
  {
    "type":"pressure",
    "data":{
      "pressure":{
        "value":-194.803,
        "timestamp":"2018-03-27 00:27:32.023314028 +0000 UTC",
        "unit":{
          "symbol":"Pa",
          "name":"pascals"
        }
      }
    },
    "location":{
      "rack":"vec1-c1-wrigley",
      "board":"vec",
      "device":"b050676303b65806ef664e6a07acf2dc"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":null,
        "timestamp":"2018-03-27 00:27:34.450837657 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec2-c1-wrigley",
      "board":"vec",
      "device":"c5d8272bc1106aa12225850d33483f33"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":18.57,
        "timestamp":"2018-03-27 00:27:34.553990408 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec2-c1-wrigley",
      "board":"vec",
      "device":"18185208cbc0e5a4700badd6e39bb12d"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":18.0,
        "timestamp":"2018-03-27 00:27:34.674081431 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec2-c1-wrigley",
      "board":"vec",
      "device":"80de4a2b73c41cf71ef7d30e1f2b9494"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.78,
        "timestamp":"2018-03-27 00:27:34.796130888 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec2-c1-wrigley",
      "board":"vec",
      "device":"fa941bad2325cb097deb97743007ff36"
    }
  },
  {
    "type":"pressure",
    "data":{
      "pressure":{
        "value":-0.95,
        "timestamp":"2018-03-27 00:27:35.401668589 +0000 UTC",
        "unit":{
          "symbol":"Pa",
          "name":"pascals"
        }
      }
    },
    "location":{
      "rack":"vec2-c1-wrigley",
      "board":"vec",
      "device":"285e7b428eb467678097ecab5f765d7c"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.78,
        "timestamp":"2018-03-27 00:27:34.270756556 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec2-c1-wrigley",
      "board":"vec",
      "device":"5beec8c6db5445fd164566478997ff30"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.41,
        "timestamp":"2018-03-27 00:27:34.363744778 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec2-c1-wrigley",
      "board":"vec",
      "device":"1e93da83dd383757474f539314446c3d"
    }
  },
  {
    "type":"pressure",
    "data":{
      "pressure":{
        "value":9.606,
        "timestamp":"2018-03-27 00:27:35.819361251 +0000 UTC",
        "unit":{
          "symbol":"Pa",
          "name":"pascals"
        }
      }
    },
    "location":{
      "rack":"vec2-c1-wrigley",
      "board":"vec",
      "device":"880d01fd6edd947ceb75f131324eecb7"
    }
  },
  {
    "type":"pressure",
    "data":{
      "pressure":{
        "value":2.85,
        "timestamp":"2018-03-27 00:27:36.244524536 +0000 UTC",
        "unit":{
          "symbol":"Pa",
          "name":"pascals"
        }
      }
    },
    "location":{
      "rack":"vec2-c1-wrigley",
      "board":"vec",
      "device":"b050676303b65806ef664e6a07acf2dc"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":18.88,
        "timestamp":"2018-03-27 00:27:35.574651911 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec2-c1-wrigley",
      "board":"vec",
      "device":"b0efdac41dfc7e8261dfd4bd2369b161"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":null,
        "timestamp":"2018-03-27 00:27:36.479001057 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec2-c1-wrigley",
      "board":"vec",
      "device":"b4bf920c24ad7c2e23360ffcda9f4b5d"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":null,
        "timestamp":"2018-03-27T00:27:36.825599+00:00",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec2-c1-wrigley",
      "board":"vec",
      "device":"3f336eb0bb188de8c72650d406225ffa"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.71,
        "timestamp":"2018-03-27 00:27:36.36390881 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec2-c1-wrigley",
      "board":"vec",
      "device":"2079f261767869c929e01ba10132eceb"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.85,
        "timestamp":"2018-03-27 00:27:36.595785182 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec2-c1-wrigley",
      "board":"vec",
      "device":"e955c02a134ff2ac6e9e3a9dedc67dc1"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.34,
        "timestamp":"2018-03-27 00:27:36.709157841 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec2-c1-wrigley",
      "board":"vec",
      "device":"645a790d97fc93ee12ec5fbb514230d7"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":18.0,
        "timestamp":"2018-03-27 00:27:32.957080791 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec5-c1-wrigley",
      "board":"vec",
      "device":"b0efdac41dfc7e8261dfd4bd2369b161"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.85,
        "timestamp":"2018-03-27 00:27:33.059350972 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec5-c1-wrigley",
      "board":"vec",
      "device":"18185208cbc0e5a4700badd6e39bb12d"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.63,
        "timestamp":"2018-03-27 00:27:33.170661964 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec5-c1-wrigley",
      "board":"vec",
      "device":"fa941bad2325cb097deb97743007ff36"
    }
  },
  {
    "type":"pressure",
    "data":{
      "pressure":{
        "value":79.853,
        "timestamp":"2018-03-27 00:27:33.779212524 +0000 UTC",
        "unit":{
          "symbol":"Pa",
          "name":"pascals"
        }
      }
    },
    "location":{
      "rack":"vec5-c1-wrigley",
      "board":"vec",
      "device":"285e7b428eb467678097ecab5f765d7c"
    }
  },
  {
    "type":"pressure",
    "data":{
      "pressure":{
        "value":48.608,
        "timestamp":"2018-03-27 00:27:35.143883189 +0000 UTC",
        "unit":{
          "symbol":"Pa",
          "name":"pascals"
        }
      }
    },
    "location":{
      "rack":"vec5-c1-wrigley",
      "board":"vec",
      "device":"880d01fd6edd947ceb75f131324eecb7"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":null,
        "timestamp":"2018-03-27T00:27:36.843812+00:00",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec5-c1-wrigley",
      "board":"vec",
      "device":"3f336eb0bb188de8c72650d406225ffa"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.71,
        "timestamp":"2018-03-27 00:27:35.243033733 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec5-c1-wrigley",
      "board":"vec",
      "device":"5beec8c6db5445fd164566478997ff30"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.49,
        "timestamp":"2018-03-27 00:27:35.32608591 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec5-c1-wrigley",
      "board":"vec",
      "device":"1e93da83dd383757474f539314446c3d"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":null,
        "timestamp":"2018-03-27 00:27:34.883335497 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec5-c1-wrigley",
      "board":"vec",
      "device":"b4bf920c24ad7c2e23360ffcda9f4b5d"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.78,
        "timestamp":"2018-03-27 00:27:35.520428113 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec5-c1-wrigley",
      "board":"vec",
      "device":"80de4a2b73c41cf71ef7d30e1f2b9494"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.56,
        "timestamp":"2018-03-27 00:27:35.636458257 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec5-c1-wrigley",
      "board":"vec",
      "device":"645a790d97fc93ee12ec5fbb514230d7"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":null,
        "timestamp":"2018-03-27 00:27:36.167096476 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec5-c1-wrigley",
      "board":"vec",
      "device":"c5d8272bc1106aa12225850d33483f33"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.85,
        "timestamp":"2018-03-27 00:27:36.271218018 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec5-c1-wrigley",
      "board":"vec",
      "device":"e955c02a134ff2ac6e9e3a9dedc67dc1"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.56,
        "timestamp":"2018-03-27 00:27:36.381432508 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec5-c1-wrigley",
      "board":"vec",
      "device":"2079f261767869c929e01ba10132eceb"
    }
  },
  {
    "type":"pressure",
    "data":{
      "pressure":{
        "value":6.65,
        "timestamp":"2018-03-27 00:27:36.057917375 +0000 UTC",
        "unit":{
          "symbol":"Pa",
          "name":"pascals"
        }
      }
    },
    "location":{
      "rack":"vec5-c1-wrigley",
      "board":"vec",
      "device":"b050676303b65806ef664e6a07acf2dc"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.56,
        "timestamp":"2018-03-27 00:27:34.485358253 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec6-c1-wrigley",
      "board":"vec",
      "device":"e955c02a134ff2ac6e9e3a9dedc67dc1"
    }
  },
  {
    "type":"pressure",
    "data":{
      "pressure":{
        "value":0.0,
        "timestamp":"2018-03-27 00:27:35.093804234 +0000 UTC",
        "unit":{
          "symbol":"Pa",
          "name":"pascals"
        }
      }
    },
    "location":{
      "rack":"vec6-c1-wrigley",
      "board":"vec",
      "device":"285e7b428eb467678097ecab5f765d7c"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.71,
        "timestamp":"2018-03-27 00:27:34.181074241 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec6-c1-wrigley",
      "board":"vec",
      "device":"b0efdac41dfc7e8261dfd4bd2369b161"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.49,
        "timestamp":"2018-03-27 00:27:34.283197766 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec6-c1-wrigley",
      "board":"vec",
      "device":"1e93da83dd383757474f539314446c3d"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":null,
        "timestamp":"2018-03-27 00:27:34.381244117 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec6-c1-wrigley",
      "board":"vec",
      "device":"c5d8272bc1106aa12225850d33483f33"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.71,
        "timestamp":"2018-03-27 00:27:36.409486619 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec6-c1-wrigley",
      "board":"vec",
      "device":"80de4a2b73c41cf71ef7d30e1f2b9494"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":null,
        "timestamp":"2018-03-27T00:27:36.879962+00:00",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec6-c1-wrigley",
      "board":"vec",
      "device":"3f336eb0bb188de8c72650d406225ffa"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.49,
        "timestamp":"2018-03-27 00:27:36.183073194 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec6-c1-wrigley",
      "board":"vec",
      "device":"5beec8c6db5445fd164566478997ff30"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.56,
        "timestamp":"2018-03-27 00:27:36.293341489 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec6-c1-wrigley",
      "board":"vec",
      "device":"2079f261767869c929e01ba10132eceb"
    }
  },
  {
    "type":"pressure",
    "data":{
      "pressure":{
        "value":0.0,
        "timestamp":"2018-03-27 00:27:33.816854809 +0000 UTC",
        "unit":{
          "symbol":"Pa",
          "name":"pascals"
        }
      }
    },
    "location":{
      "rack":"vec6-c1-wrigley",
      "board":"vec",
      "device":"b050676303b65806ef664e6a07acf2dc"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.56,
        "timestamp":"2018-03-27 00:27:36.522880402 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec6-c1-wrigley",
      "board":"vec",
      "device":"fa941bad2325cb097deb97743007ff36"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.56,
        "timestamp":"2018-03-27 00:27:36.658689779 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec6-c1-wrigley",
      "board":"vec",
      "device":"645a790d97fc93ee12ec5fbb514230d7"
    }
  },
  {
    "type":"pressure",
    "data":{
      "pressure":{
        "value":0.0,
        "timestamp":"2018-03-27 00:27:36.895690617 +0000 UTC",
        "unit":{
          "symbol":"Pa",
          "name":"pascals"
        }
      }
    },
    "location":{
      "rack":"vec6-c1-wrigley",
      "board":"vec",
      "device":"880d01fd6edd947ceb75f131324eecb7"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.63,
        "timestamp":"2018-03-27 00:27:33.983691282 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec6-c1-wrigley",
      "board":"vec",
      "device":"18185208cbc0e5a4700badd6e39bb12d"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":null,
        "timestamp":"2018-03-27 00:27:34.097590495 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec6-c1-wrigley",
      "board":"vec",
      "device":"b4bf920c24ad7c2e23360ffcda9f4b5d"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":null,
        "timestamp":"2018-03-27 00:27:33.475447487 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec3-c1-wrigley",
      "board":"vec",
      "device":"c5d8272bc1106aa12225850d33483f33"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.56,
        "timestamp":"2018-03-27 00:27:33.843729447 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec3-c1-wrigley",
      "board":"vec",
      "device":"80de4a2b73c41cf71ef7d30e1f2b9494"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.49,
        "timestamp":"2018-03-27 00:27:33.598833658 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec3-c1-wrigley",
      "board":"vec",
      "device":"fa941bad2325cb097deb97743007ff36"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.41,
        "timestamp":"2018-03-27 00:27:33.714140935 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec3-c1-wrigley",
      "board":"vec",
      "device":"645a790d97fc93ee12ec5fbb514230d7"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":null,
        "timestamp":"2018-03-27T00:27:36.919100+00:00",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec3-c1-wrigley",
      "board":"vec",
      "device":"3f336eb0bb188de8c72650d406225ffa"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.85,
        "timestamp":"2018-03-27 00:27:33.930279806 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec3-c1-wrigley",
      "board":"vec",
      "device":"b0efdac41dfc7e8261dfd4bd2369b161"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.49,
        "timestamp":"2018-03-27 00:27:34.039614426 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec3-c1-wrigley",
      "board":"vec",
      "device":"2079f261767869c929e01ba10132eceb"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":null,
        "timestamp":"2018-03-27 00:27:34.150667279 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec3-c1-wrigley",
      "board":"vec",
      "device":"b4bf920c24ad7c2e23360ffcda9f4b5d"
    }
  },
  {
    "type":"pressure",
    "data":{
      "pressure":{
        "value":5.858,
        "timestamp":"2018-03-27 00:27:34.388039226 +0000 UTC",
        "unit":{
          "symbol":"Pa",
          "name":"pascals"
        }
      }
    },
    "location":{
      "rack":"vec3-c1-wrigley",
      "board":"vec",
      "device":"880d01fd6edd947ceb75f131324eecb7"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.56,
        "timestamp":"2018-03-27 00:27:35.927244334 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec3-c1-wrigley",
      "board":"vec",
      "device":"5beec8c6db5445fd164566478997ff30"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.56,
        "timestamp":"2018-03-27 00:27:34.567299471 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec3-c1-wrigley",
      "board":"vec",
      "device":"1e93da83dd383757474f539314446c3d"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":18.8,
        "timestamp":"2018-03-27 00:27:34.681450265 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec3-c1-wrigley",
      "board":"vec",
      "device":"18185208cbc0e5a4700badd6e39bb12d"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.93,
        "timestamp":"2018-03-27 00:27:34.794535662 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec3-c1-wrigley",
      "board":"vec",
      "device":"e955c02a134ff2ac6e9e3a9dedc67dc1"
    }
  },
  {
    "type":"pressure",
    "data":{
      "pressure":{
        "value":7.336,
        "timestamp":"2018-03-27 00:27:35.221170999 +0000 UTC",
        "unit":{
          "symbol":"Pa",
          "name":"pascals"
        }
      }
    },
    "location":{
      "rack":"vec3-c1-wrigley",
      "board":"vec",
      "device":"b050676303b65806ef664e6a07acf2dc"
    }
  },
  {
    "type":"pressure",
    "data":{
      "pressure":{
        "value":12.983,
        "timestamp":"2018-03-27 00:27:35.836262824 +0000 UTC",
        "unit":{
          "symbol":"Pa",
          "name":"pascals"
        }
      }
    },
    "location":{
      "rack":"vec3-c1-wrigley",
      "board":"vec",
      "device":"285e7b428eb467678097ecab5f765d7c"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.85,
        "timestamp":"2018-03-27 00:27:36.371482079 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec4-c1-wrigley",
      "board":"vec",
      "device":"b0efdac41dfc7e8261dfd4bd2369b161"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.71,
        "timestamp":"2018-03-27 00:27:36.475687036 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec4-c1-wrigley",
      "board":"vec",
      "device":"1e93da83dd383757474f539314446c3d"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":20.06,
        "timestamp":"2018-03-27 00:27:36.579821269 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec4-c1-wrigley",
      "board":"vec",
      "device":"e955c02a134ff2ac6e9e3a9dedc67dc1"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.78,
        "timestamp":"2018-03-27 00:27:36.708051003 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec4-c1-wrigley",
      "board":"vec",
      "device":"80de4a2b73c41cf71ef7d30e1f2b9494"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.78,
        "timestamp":"2018-03-27 00:27:36.831144507 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec4-c1-wrigley",
      "board":"vec",
      "device":"fa941bad2325cb097deb97743007ff36"
    }
  },
  {
    "type":"pressure",
    "data":{
      "pressure":{
        "value":21.85,
        "timestamp":"2018-03-27 00:27:35.666520621 +0000 UTC",
        "unit":{
          "symbol":"Pa",
          "name":"pascals"
        }
      }
    },
    "location":{
      "rack":"vec4-c1-wrigley",
      "board":"vec",
      "device":"b050676303b65806ef664e6a07acf2dc"
    }
  },
  {
    "type":"pressure",
    "data":{
      "pressure":{
        "value":104.658,
        "timestamp":"2018-03-27 00:27:36.276649856 +0000 UTC",
        "unit":{
          "symbol":"Pa",
          "name":"pascals"
        }
      }
    },
    "location":{
      "rack":"vec4-c1-wrigley",
      "board":"vec",
      "device":"285e7b428eb467678097ecab5f765d7c"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.71,
        "timestamp":"2018-03-27 00:27:36.918600792 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec4-c1-wrigley",
      "board":"vec",
      "device":"5beec8c6db5445fd164566478997ff30"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":18.57,
        "timestamp":"2018-03-27 00:27:33.710106794 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec4-c1-wrigley",
      "board":"vec",
      "device":"2079f261767869c929e01ba10132eceb"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":20.84,
        "timestamp":"2018-03-27 00:27:31.329975591 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec4-c1-wrigley",
      "board":"vec",
      "device":"18185208cbc0e5a4700badd6e39bb12d"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":null,
        "timestamp":"2018-03-27 00:27:31.442163201 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec4-c1-wrigley",
      "board":"vec",
      "device":"b4bf920c24ad7c2e23360ffcda9f4b5d"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":null,
        "timestamp":"2018-03-27T00:27:36.969472+00:00",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec4-c1-wrigley",
      "board":"vec",
      "device":"3f336eb0bb188de8c72650d406225ffa"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":null,
        "timestamp":"2018-03-27 00:27:34.862950819 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec4-c1-wrigley",
      "board":"vec",
      "device":"c5d8272bc1106aa12225850d33483f33"
    }
  },
  {
    "type":"temperature",
    "data":{
      "temperature":{
        "value":17.71,
        "timestamp":"2018-03-27 00:27:35.000592161 +0000 UTC",
        "unit":{
          "symbol":"C",
          "name":"degrees celsius"
        }
      }
    },
    "location":{
      "rack":"vec4-c1-wrigley",
      "board":"vec",
      "device":"645a790d97fc93ee12ec5fbb514230d7"
    }
  },
  {
    "type":"pressure",
    "data":{
      "pressure":{
        "value":-71.356,
        "timestamp":"2018-03-27 00:27:35.247448539 +0000 UTC",
        "unit":{
          "symbol":"Pa",
          "name":"pascals"
        }
      }
    },
    "location":{
      "rack":"vec4-c1-wrigley",
      "board":"vec",
      "device":"880d01fd6edd947ceb75f131324eecb7"
    }
  }
]vapor@vec1-c1-wrigley:~/src```